File: 2015/javascript/day_01_2015.js
const fs = require("fs");
const data = fs.readFileSync("day_01_2015_input.txt", "utf-8");

const floors = (data) => {
  let floor = 0;

  for (let i = 0; i < data.length; i++) {
    if (data[i] === "(") {
      floor++;
    } else {
      floor--;

      if (floor < 0) {
        return i + 1;
      }
    }
  }
  return floor;
};

console.log(floors(data)); // 1783

File: 2015/python/day_01_2015.py
# day_01_2015.py - https://adventofcode.com/2015/day/1


import os
from pathlib import Path


def floors(data):
    floor = 0
    data = tuple(data)
    for index, char in enumerate(data):
        if char == "(":
            floor += 1
        else:
            floor -= 1
            if floor < 0:
                return index + 1
    return floor


def main():
    os.chdir(Path(__file__).parent)
    with open("day_01_2015_input.txt", "r") as infile:
        data = infile.read()
    print(floors(data))


if __name__ == "__main__":
    main() # 1783
